zilliqa/src/exec.rs

```diff
@@ -444,11 +446,23 @@ impl State {
             }
         }
 
+        // If this was contract creation, apply this to the deltas for the convenience of
```

Comment:
I'm confused how this would only be for the convenience of the caller? Would something later result in this happening anyway?

Reply:
Good question, this comment was already there in the code so I'm not sure. My best guess is that it reduces the need to special case contract creation in the layer above, by letting most of it be handled by `apply_deltas()` rather than needing dedicated logic?
@n-hutton might know more about this.